### PR TITLE
Packaging Suggestions

### PR DESCRIPTION
**Pull request by Cadair**
:wave: Hope you don't mind but I thought I would take a pass over your packaging infrastructure. This does the following things:

1. Move most of the variables in `setup.py` into `setup.cfg` as a more modern place to put them.
2. Uses `pyproject.toml` to ensure that `numpy` is installed before `setup.py` is run.
3. Removes the requirements files in favour of setuptools extras requires. (It is generally recommended that `requirements.txt` files are used for applications and not libraries. There is no need for them to be removed of course, but extras are good to have as well.)
4. Adds a `tox.ini` file which can run the tests and build the docs. It also tests against the installed version of the sdist to ensure that the compiled extensions are copied properly.
5. Swaps from `numpydoc` to the built in `sphinx.ext.napoleon`, they are almost identical, and it doesn't look like you are running into any of the edge cases.
6. Refactors your azure pipelines file to use https://github.com/OpenAstronomy/azure-pipelines-templates and enable building wheels (mainly to see if it works).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,76 +1,42 @@
-# Python package
-# Create and test a Python package on multiple Python versions.
-# Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
-# https://docs.microsoft.com/azure/devops/pipelines/languages/python
+resources:
+  repositories:
+  - repository: OpenAstronomy
+    type: github
+    endpoint: github
```
Comment (Cadair) [inline azure-pipelines.yml]: I need to know what your azure github endpoint is called. i.e. under project settings, here it's "sunpy":

image

**Comment by Cadair** [comment]
This can also easily be extended to push releases to PyPI btw, see the readme on https://github.com/OpenAstronomy/azure-pipelines-templates

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,76 +1,42 @@
...
+    - '*dev*'
+    - '*pre*'
+    - '*post*'
+
+jobs:
+- template: run-tox-env.yml@OpenAstronomy
+  parameters:
+    libraries:
+      apt: gfortran
+      brew: gfortran
```
Comment (Cadair) [inline azure-pipelines.yml]: Is the reason you were not building for windows not being able to install gfortran?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,76 +1,50 @@
...
-# Create and test a Python package on multiple Python versions.
-# Add steps that analyze code, save the dist with the build record, publish to a PyPI-compatible index, and more:
-# https://docs.microsoft.com/azure/devops/pipelines/languages/python
+name: $(BuildDefinitionName)_$(Date:yyyyMMdd)$(Rev:.rr)
+variables:
+  CI_NAME: Azure Pipelines
+  CI_BUILD_ID: $(Build.BuildId)
+  CI_BUILD_URL: "https://dev.azure.com/dstansby/streamtracer/_build/results?buildId=$(Build.BuildId)"
+  CIBW_BUILD: cp36-* cp37-* cp38-*
+  CIBW_SKIP: "*-win32 *-manylinux1_i686"
```
Comment (Cadair) [inline azure-pipelines.yml:7]: I skipped windows wheels here for now, but we can try and make them build.

File: azure-pipelines.yml (hunk 1 above)
Comment (dstansby) [reply azure-pipelines.yml]: It's 'dstansby'

File: azure-pipelines.yml (hunk 2 above)
Comment (dstansby) [reply azure-pipelines.yml]: Not quite, I can install gfortran, but can't get it discovered when the fortran tries to be copiled.

File: azure-pipelines.yml (hunk 3 above)
Comment (dstansby) [reply azure-pipelines.yml:7]: Maybe I'll open another PR for this

File: azure-pipelines.yml (hunk 1 above)
Comment (dstansby) [reply azure-pipelines.yml]: Added this manually

**Comment by dstansby** [comment]
Ahhhhhh I think I broke the azure connection again

**Comment by dstansby** [comment]
@Cadair got it working again, but it is not happy 😢 : https://dev.azure.com/dstansby/streamtracer/_build/results?buildId=161&view=results

**Comment by Cadair** [comment]
That error is referencing this: https://github.com/OpenAstronomy/azure-pipelines-templates/blob/master/run-tox-env.yml#L66 but I can't see what the issue is...

I wonder if @astrofrog has any bright ideas?!

**Comment by astrofrog** [comment]
Looks like @dstansby already fixed the issue and now needs a fix for https://github.com/OpenAstronomy/azure-pipelines-templates/issues/31 :)

**Comment by Cadair** [comment]
wat :man_facepalming: eugh, azure UI is a pain.

**Comment by dstansby** [comment]
🎉 progress!

**Comment by Cadair** [comment]
looks like you are missing the fortran compiler for testing on the sdist and on the macos wheel build

**Comment by dstansby** [comment]
I thought I added them just now, but it might be that the `publish.yml` thingy doesn't recognise dependency installation at the moment.

**Comment by dstansby** [comment]
Thanks for opening this! I think I've now implemented most of it.

**Pull request closed without merging**